### PR TITLE
TodoList

### PR DESCRIPTION
**Pull request by MaximMatveenko**

### CODE REVIEW

File: src/TodoList.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    addItem(e) {
+      if (this._inputElement.value !== "") {
+        var newItem = {
+          text: this._inputElement.value,
+          key: Date.now()
+        };
+  
+        this.setState((prevState) => {
+          return {
+            items: prevState.items.concat(newItem)
```
Comment (prodislav) [inline src/TodoList.js]: use spread operator here

File: src/TodoItems.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class TodoItems extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.createTasks = this.createTasks.bind(this);
+    }
+
+    createTasks(item) {
+        return <li key={item.key}>{item.text}<button onClick ={() => this.delete(item.key)} className="taskDelBtn">X</button></li>
```
Comment (prodislav) [inline src/TodoItems.js]: format this row

File: src/TodoItems.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    delete(key) {
+        this.props.delete(key);
+    }
+
+    // edit: function () {
+        
+    // }
```
Comment (prodislav) [inline src/TodoItems.js]: try by hard u can do it

File: src/TodoItems.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    render() {
+        var todoEntries = this.props.entries;
+        var listItems = todoEntries.map(this.createTasks);
```
Comment (prodislav) [inline src/TodoItems.js]: use let or const

File: src/TodoItems.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+class TodoItems extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.createTasks = this.createTasks.bind(this);
+    }
+
+    createTasks(item) {
```
Comment (prodislav) [inline src/TodoItems.js]: u can try to use arrow function without bind in constructor

File: src/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      this.state = {
+        items: []
+      };
+  
+      this.addItem = this.addItem.bind(this);
+      this.deleteItem = this.deleteItem.bind(this);
+    }
+  
+    addItem(e) {
+      if (this._inputElement.value !== "") {
```
Comment (prodislav) [inline src/TodoList.js]: image
try to use this instead ref

File: src/index.css [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    font-size: 40px;
+    padding: 10px;
+    margin-left: 20px;
+}
+
+.TaskInput {
+    padding: 20px;
+}
+
+#TasksTextInput {
```
Comment (prodislav) [inline src/index.css]: rewrite styles using bem

File: src/TodoItems.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+        return (
+            <ul className="theList">
+                {listItems}
+            </ul>
+        )
+    }
+}
+
+export default TodoItems;
```
Comment (prodislav) [inline src/TodoItems.js]: u can try to use export without default. google help u

File: src/index.css (hunk 7 above)
Comment (MaximMatveenko) [reply src/index.css]: complete

File: src/TodoItems.js (hunk 4 above)
Comment (MaximMatveenko) [reply src/TodoItems.js]: complete

File: src/TodoItems.js (hunk 5 above)
Comment (MaximMatveenko) [reply src/TodoItems.js]: complete

File: src/TodoItems.js (hunk 8 above)
Comment (MaximMatveenko) [reply src/TodoItems.js]: complete

File: src/TodoItems.js [hunk 9]
```diff
@@ -3,22 +3,26 @@ import React, { Component } from 'react';
 export class TodoItems extends Component {
 
     createTasks = (item) => {
```
Comment (prodislav) [inline src/TodoItems.js]: createTask

File: src/TodoItems.js [hunk 10]
```diff
@@ -3,22 +3,26 @@ import React, { Component } from 'react';
...
     }
 
     delete(key) {
         this.props.delete(key);
     }
 
     render() {
-        let todoEntries = this.props.entries;
-        let listItems = todoEntries.map(this.createTasks);
+        let listItems = this.props.entries.map(this.createTasks);
```
Comment (prodislav) [inline src/TodoItems.js]: const

File: src/TodoList.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      <div className="App">
+        <div className="legacy">
+          ToDo:
+        </div>
+        <TodoItems entries={this.state.items} delete={this.deleteItem} />
+        <hr></hr>
+        <div className="TaskInput">
+          <form onSubmit={this.addItem}>
+            <input ref={(node) => { this._inputElement = node }}
+              id="TasksTextInput"
```
Comment (prodislav) [inline src/TodoList.js]: id?

File: src/TodoList.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    this._inputElement.value = "";
+
+    console.log(this.state.items);
+
+    e.preventDefault();
+  }
+
+
+  deleteItem = (key) => {
+    let filteredItem = this.state.items.filter(function (item) {
```
Comment (prodislav) [inline src/TodoList.js]: filteredItems

**Pull request merged**